.jenkins/benchmark/run-benchmark.sh

```diff
@@ -1,43 +1,110 @@
 #!/bin/bash
 
+__fp8="no"
+
+while [ -n "$1" ];
+    do
+        case $1 in
+        -fp8 )
+            __fp8="yes"
+            ;;
+        esac
+        shift
+    done
+
 model=/mnt/weka/data/pytorch/llama3.1/Meta-Llama-3.1-8B
 model_short=$(basename $model)
 #replace all '.' with '-'
 model_short=${model_short//./-}
-log_file=$(mktemp /tmp/_benchmark_${model_short}_XXXXXXX.log)
+tmp_file_name=$(mktemp /tmp/_benchmark_${model_short}_XXXXXXX)
+error_log_file="${tmp_file_name}_error.log"
+log_file="${tmp_file_name}.log"
+
+scenario=fp8
+if [[ $__fp8 == "no" ]]; then
+    scenario=bf16
+fi
 
 # Generate an empty result file.
 # This way in case of any crash it will be treated by jenkins as failure
 if [[ -n "$TEST_RESULTS_DIR" ]]; then
     mkdir -p ${TEST_RESULTS_DIR}
-    LOG_PATH=$(mktemp ${TEST_RESULTS_DIR}/benchmark_${model_short}_XXXXXX.xml)
+    LOG_PATH=$(mktemp ${TEST_RESULTS_DIR}/benchmark_${model_short}_${scenario}_XXXXXX.xml)
 fi
 
-throughput_threshold=999999 
-if [[ "${PERF_THRESHOLD}" ]]; then
-    throughput_threshold=${PERF_THRESHOLD}
-fi
-warmup_threshold=1 
-if [[ "${WARMUP_THRESHOLD}" ]]; then
-    warmup_threshold=${WARMUP_THRESHOLD}
-fi
+
+# Get threshold values according to the scenario and env variables
+throughput_threshold=999999
+
+IFS=',' read -ra pairs <<< "$PERF_THRESHOLD"
+for pair in "${pairs[@]}"; do
+    IFS='=' read -ra kv <<< "$pair"
+    key="${kv[0]}"
+    value="${kv[1]}"
+
+    if [[ $key == $scenario ]]; then
+        throughput_threshold=$value
+    fi
+done
+
+warmup_threshold=1
+
+IFS=',' read -ra pairs <<< "$WARMUP_THRESHOLD"
+for pair in "${pairs[@]}"; do
+    IFS='=' read -ra kv <<< "$pair"
+    key="${kv[0]}"
+    value="${kv[1]}"
+
+    if [[ $key == $scenario ]]; then
+        warmup_threshold=$value
+    fi
+done
 
 # Get the directory of the current script
 script_dir=$(dirname "$(readlink -f "$0")")
 
 start=`date +%s`
-python  $script_dir/../../benchmarks/benchmark_throughput.py \
-    --model $model \
-    --device hpu \
-    --seed 2024 \
-    --backend vllm \
-    --dataset /mnt/weka/data/pytorch/llama2/ShareGPT_V3_unfiltered_cleaned_split.json \
-    --num-prompts 1000 \
-    --dtype bfloat16 \
-    --max-model-len 4096 \
-    --max-num-batched-tokens 8192 \
-    --max-num-seqs 128 \
-    --use-padding-aware-scheduling |& tee $log_file
+
+export TQDM_DISABLE=True
+
+if [[ $__fp8 == "yes" ]]; then
```

Comment:
I am not stick to that, but maybe we can use the same call to benchmark and only add additional parameter to fp8?
I thought that for fp8 we need extra preparation - if yes this comment will be not relevant